src/Shared/ValueStringBuilder/ValueListBuilder.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+// Copied from https://github.com/dotnet/runtime/blob/a9ed4168626c14b4d74db0d8c205c69e56fc45ed/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ValueListBuilder.cs
```

Comment:
Is this covered by one of our sync processes or is it a true copy?

Reply:
True copy of a portion of the type

Reply:
Well, I look forward to this getting out of sync with the real one, but you have to do what you have to do for perf...